File: linked-list/deleteDuplicates.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * 删除排序链表中的重复元素
 * 给定一个排序链表，删除所有重复的元素，使得每个元素只出现一次。
 * 输入: 1->1->2->3->3 输出: 1->2->3
 * 
 * @param {*} head 
 */
var deleteDuplicates = function(head) {
    let current_node = head

    while (current_node) {
        let cur_val = current_node.val
        let pre_node = current_node.next

        if(cur_val != pre_node.val){
            current_node = pre_node
        }else{
            // 相等时，改变指针，再入循环
            current_node.next = pre_node.next
        }

    }
};

File: linked-list/deleteDuplicates.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:

        cur_node = head

        while cur_node and cur_node.next:
            
            pre_node = cur_node.next

            if cur_node.val == pre_node.val:
                cur_node.next = pre_node.next
            else:
                cur_node = pre_node
        
        return head